contracts/staking/StakingRewardsCampaign.sol

```diff
@@ -0,0 +1,168 @@
+
+    // User address => Lock Type => Staked Information
+    mapping(address => mapping(bytes32 => StakeInfo)) private stakeInfo;
+    mapping(bytes32 => LockInfo) public lockPeriod;
+
+    /* ========== Initializer ========== */
+
+    function initialize(address _stakingToken) public initializer {
+        __Ownable_init_unchained();
+        __Pausable_init_unchained();
+        __ReentrancyGuard_init_unchained();
+        __StakingRewards_init_unchained(_stakingToken);
+    }
+
+    function __StakingRewards_init_unchained(address _stakingToken)
+        internal
+        onlyInitializing
+    {
+        stakingToken = IERC20Upgradeable(_stakingToken);
+    }
+
+    /* ========== VIEWS ========== */
+
+    // Staked Amount of MEE token into specific Raffle or the second staking optioin
+    function balanceOf(address account, bytes32 lockType)
+        external
+        view
+        returns (uint256)
+    {
+        return stakeInfo[account][lockType].balance;
+    }
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+
+    function stake(uint256 amount, bytes32 lockType)
+        external
+        nonReentrant
+        whenNotPaused
+    {
+        LockInfo memory lockInfo = lockPeriod[lockType];
+        // In case of Staking Option 1(Raffle), amount just represents the ticket amount
+        // In case of Staking Option 2(Land), we don't use the amount variable but it should be 1 always
+        require(amount >= 1, "Invalid Amount");
+        require(lockInfo.minAmount != 0, "Not Active");
+        require(amount * lockInfo.minAmount <= lockInfo.maxAmount || lockInfo.maxAmount == 0, "Invalid Amount");
+
+        // Stake Information for the specific Staking Option for the user. Raffle or Land Option
+        StakeInfo storage info = stakeInfo[msg.sender][lockType];
+
+        info.balance = info.balance + lockInfo.minAmount * amount;
+        info.unlockTime = uint64(block.timestamp) + lockInfo.period;
+
+        // Lock MEE tokens into the staking contract
+        stakingToken.safeTransferFrom(
+            msg.sender,
+            address(this),
+            lockInfo.minAmount * amount
+        );
+
+        emit Staked(
+            msg.sender, // Owner
+            lockInfo.minAmount * amount, // Token Amount
+            amount, // Ticket Amount for Staking Option 1 or 1 for Staking Option 2
+            lockType, // Lock Type
+            info.unlockTime
+        );
+    }
+
+    function withdraw(bytes32 lockType)
+        public
+        nonReentrant
+        whenNotPaused
+        lockPeriodCheck(lockType)
+    {
+        StakeInfo storage info = stakeInfo[msg.sender][lockType];
+        // Calcaulate the balance to withdraw for the specific Land Tier and the Staking Option
+        uint256 balance = info.balance;
+
+        require(balance > 0, "Nothing to withdraw");
+        // Unlock Mee Token after lock period.
+        stakingToken.safeTransfer(msg.sender, balance);
+
+        delete stakeInfo[msg.sender][lockType];
```

Comment:
I recommend swapping lines 106 and 108, otherwise an attack with specifically low gas could possibly be made to terminate in line 107 and exfiltrate unlimited tokens.